fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java

```diff
@@ -77,6 +83,10 @@
 
     private static final Logger LOGGER = getLogger(HttpResourceConverter.class);
 
+    // Regex pattern which decomposes a http resource uri into fcr components
+    final static Pattern FORWARD_COMPONENT_PATTERN = Pattern.compile(
```

Comment:
Please make this `private`.

Also, a textual description of what is going on in the regex would be helpful for future debugging.

Also, the regex has four capture groups, but only three appear to be used in the code. Is this intentional?

Reply:
Should be fixed now.  Originally the fourth capture group was to differentiate between mementos and timemaps, but since those aren't being handled differently in the code currently the parenthetical block is just for treating the whole memento id and preceding slash as a single optional token at the end of the path.  So its intentional, just not serving as a group.